src/app/favourite/favourite.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+import { FirebaseService } from '../firebase.service';
+
+@Component({
+  selector: 'app-favourite',
+  templateUrl: './favourite.component.html',
+  styleUrls: ['./favourite.component.scss']
+})
+export class FavouriteComponent implements OnInit {
+  recipe: any;
+  btnColor = "#f3f3f3";
+
+  constructor(private fbService: FirebaseService) { }
+
+  ngOnInit(): void {
+    this.fbService.getRecipe(0).valueChanges().subscribe(recipe => {
```

Comment:
why "valueChanges"?

Reply:
add the type to recipe. if you're not sure what type it is, at least add "any". (recipe: any) =>